flexmeasures/data/models/planning/__init__.py

```diff
@@ -34,26 +36,38 @@ class Scheduler:
     __version__ = None
     __author__ = None
 
-    sensor: Sensor
+    sensor: Optional[Sensor] = None
+    asset: Optional[Asset] = None
+
+    asset_or_sensor: Asset | Sensor
+
     start: datetime
     end: datetime
     resolution: timedelta
     belief_time: datetime
+
     round_to_decimals: int
+
     flex_model: Optional[dict] = None
     flex_context: Optional[dict] = None
+
     fallback_scheduler_class: "Type[Scheduler] | None" = None
     info: dict | None = None
 
     config_deserialized = False  # This flag allows you to let the scheduler skip checking config, like timing, flex_model and flex_context
     return_multiple = False
 
+    # set to True if the Scheduler supports triggering on an Asset or False
+    # if the Scheduler expects a Sensor
+    requires_asset = False
+
     def __init__(
         self,
-        sensor,
-        start,
-        end,
-        resolution,
+        asset_or_sensor: Asset | Sensor | None = None,
+        start: Optional[datetime] = None,
+        end: Optional[datetime] = None,
+        resolution: Optional[timedelta] = None,
+        sensor: Optional[Sensor] = None,
```

Comment:
As a class init parameter, I'd like to see this remain in the first position. Otherwise, it is breaking code that passes the sensor as an arg rather than a kwarg.

We should also add an inline comment telling developers that this argument is deprecated:
```
sensor: Optional[Sensor] = None,  # deprecated
```

Reply:
Good point. At some point we'll move the asset_or_sensor to the first position and drop the sensor_id argument